src/docs/guides/database-migration-guide.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Migrating from Legacy Plugins to Database Services
+---
+
+We've introduced a more secure and flexible way to deploy databases on Railway.  The purpose of this guide is to provide an overview of why we are making these changes, what they mean for you, and how you can migrate.
+
+## What is Changing?
+### Legacy Plugins
+In the early stages of Railway, we introduced `plugins`, a foundational part of our offering. They were databases deployed from Docker images with a fixed version, no data persistence, and no way to customize the database to your needs.
+
+### Database Services
+Some of our newest features - Volumes, Docker Image Deployments, TCP Proxy - open the door for deploying databases as services, complete with data persistence, flexible configuration and management, and connectivity from the external network.
+
+## Why Migrate?
+
+There are several primary reasons why this migration is beneficial -
+
+1. Data Security
+2. Modern Architecture
+3. Flexibility in configuration
+
+## Migration Process
+
+Here is what will happen when you choose to migrate via our one-click migration process - 
+
+  *screenshot of button?*
+
+1. **One-click button** - When you access the plugin settings from within your project canvas, you will see a prompt to migrate.
+2. **Modal** - Once you click on the button, a modal will appear, detailing the steps that will be taken on your behalf to migrate your data.
+3. **Services Stop** - Any service within your project that is connected via a variable reference to the plugin will be stopped for the duration of the migration to prevent data loss or corruption.
+4. **Database Service** - A new database service with an attached volume will be deployed.
+5. **Data Transfer** - A separate service will be deployed that will execute a migration script.  The script will connect to your old plugin, create a dump of the data, and then transfer it to the new database service.
+6. **Reference Update** - All variable references within other services that point to the old plugin will be updated to the new database service.
+
+_**NOTE: Your old plugin will NOT be deleted automatically.  You should perform the necessary actions to ensure data consistency between the plugin and the new database service.**_
```

Comment:
Should also mention that the user needs to delete the plugin when the migration is completed